### PR TITLE
New auto-layout of space views

### PR DESCRIPTION
**Pull request by emilk**
Closes https://github.com/rerun-io/rerun/issues/2139

### What
This simplifies the auto-layout of space views.

We make heavy use of `egui_tiles::Grid`, which is responsive to the current viewport size. This means that space views may move around as you resize window in an effort to keep each one square-ish. In the future we can make `egui_tiles` smarter here and inform it of the desired aspect ratio of each space view.

We make an exception for the (common) case of exactly three space views, which doesn't fit nicely in a grid anyways.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested https://demo.rerun.io/pr/2558 

PR Build Summary: https://build.rerun.io/pr/2558

Docs preview: https://rerun.io/preview/e2ed87c/docs
Examples preview: https://rerun.io/preview/e2ed87c/examples

### Testing
In order to test this I used the `run_all.py` script and found a bunch of issues with is, so I fixed them.

image

image

image

image

image

image

### CODE REVIEW

File: crates/re_viewport/src/auto_layout.rs [hunk 1]
```diff
@@ -60,234 +33,86 @@ pub(crate) fn tree_from_space_views(
...
+    // |          |            |
+    // |          |            |
+    // |          |            |
+    // +----------+------------+
+    //
+    // But which space gets a full side, and which doesn't?
+    // Answer: we prioritize them by category:
+
+    /// lower is better
+    fn category_priority(category: ViewCategory) -> usize {
```
Comment (Wumpf) [inline crates/re_viewport/src/auto_layout.rs:98]: quite unhappy to see this new use of `ViewCategory` when I'm still trying to get rid of it (not finding the time between all the higher prioritized things). Can you instead introduce a field on `SpaceViewClass` and/or replace the aspect ratio field we have there?

**Review by Wumpf — APPROVED** [review]
great step into the right direction. It does shrink customizability a little bit for the moment since the grid can't be customized easily - one can loose columns or rows but not restablish them.

I'd be very happy if you remove the new/changed use of view category, but not blocking over this ofc.

thanks for fixing up run_all properly as well, admittedly I ended up working around or giving up on it instead of taking the time to fix it up

File: crates/re_viewport/src/auto_layout.rs (hunk 1 above)
Comment (emilk) [reply crates/re_viewport/src/auto_layout.rs:98]: I want to keep aspect ratio for future use (when integrated into egui_tiles at some theoretical point in time), but yeah, I can add something like `LayoutPriority` or something

Comment (emilk) [reply crates/re_viewport/src/auto_layout.rs]: Fixed!

**Review by Wumpf — APPROVED** [review]
thanks!

**Pull request merged**